pandas/core/indexing.py

```diff
@@ -2016,6 +2016,8 @@ def _setitem_single_column(self, loc: int, value, plane_indexer):
                 and (
                     np.shares_memory(new_values, orig_values)
                     or new_values.shape != orig_values.shape
+                    or not can_hold_element(orig_values, np.nan)
+                    and isna(new_values).any()
```

Comment:
can you add parens so i dont have to remember the order of and/or operations

Reply:
sure, done